### ISSUE TITLE
travis error AssertionError from coverage/config.py in one of our test files

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbemtools#21]
Fix for m2crypto revealed another issue that shows itself in travis as:
```
    def test(self):  # pylint: disable=missing-docstring
        command = 'pywbemcli %s --help' % (cmd_str)
        args = shlex.split(command)
        proc = Popen(args, stdout=PIPE, stderr=PIPE)
        out, err = proc.communicate()
        exitcode = proc.returncode   
        self.assertEqual(exitcode, 0, '%s: ExitCond Err, cmd="%s" '
                         'exitcode %s' % (tname, command, exitcode))
        self.assertEqual(err, "", '%s stderr not empty. returned %s'
>                        % (tname, err))

E       AssertionError: test_qualget stderr not empty. returned /home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/coverage/config.py:29: DeprecationWarning: You passed a bytestring as `filenames`. This will not work on Python 3. Use `cp.read_file()` or switch to using Unicode strings across the board.

E         return configparser.RawConfigParser.read(self, filename, **kwargs)
```
It is coming from the processing in one of the tests where we execute calls with help on pywbemcli itself. I suggest that we commit the m2crypto change and I will sort out the other issue. Completed and closed the m2crypto pr but kept the issue open since the solution depends on taking from the the git for m2crypto, not pypy\i

Quick search shows the following bug from coverage.py

https://bitbucket.org/ned/coveragepy/issues/530/deprecationwarning-you-passed-a-bytestring

They say it is fixed for coverage 4.3.4 which is what we are using so not sure what the issue is so far.

**Comment by KSchopmeyer** [comment]
Won't Fix.  Mark closed

**Comment by KSchopmeyer** [comment]
I reopened this as an issue.  We are still getting it and still not sure why other than it is apparently from coverage.

Here is another example:

 err /home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/coverage/config.py:29: DeprecationWarning: You passed a bytestring as `filenames`. This will not work on Python 3. Use `cp.read_file()` or switch to using Unicode strings across the board.

  return configparser.RawConfigParser.read(self, filename, **kwargs)

 exitcode 0

 len(err) 323

err /home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/coverage/config.py:29: DeprecationWarning: You passed a bytestring as `filenames`. This will not work on Python 3. Use `cp.read_file()` or switch to using Unicode strings across the board.

  return configparser.RawConfigParser.read(self, filename, **kwargs)

**Comment by KSchopmeyer** [comment]
In discussion Friday 1 Apr we concluded that there is really a bigger issue here, the fact that deprecation warnings can appear in our testing at any time and confuse the whole test.  The question is whether we can disable the deprecation warnings for our tests and also should we.

**Comment by KSchopmeyer** [comment]
Found that this is a fairly special problem in that the issue is coming from a tool we are using (coverage) and the fact that that tool hits a condition in an import (subprocess) that causes the tools code to generate the compatibility warning.  Finally since we are testing the execution of pywbemcli using subprocess, the coverage warning gets output from pywbemcli running as a process as stderr when normally the command should NOT be issuing any stderr.

Even more, the issue was fixed in coverage in v 4.3 and when we setup we install version 4.3.4.  However, later in the .travis.yml processes we pip coveralls and that in turn reinstalls coverage but installs an old version.

In short our normal workarounds will not work.  We can either bypass the code causing the problem (i.e. one test for python 2.7) or force a fix to coverage.   Proposal is to simply fix .travis.yml after the pip install of coveralls to do an upgrade to coverage bringing it back to the latest version.

**Comment by andy-maier** [comment]
Fixed in PR #29.
